lib/directory-tree.js

```diff
@@ -27,16 +27,18 @@ function directoryTree (path, options, onEachFile) {
 
 	try { stats = FS.statSync(path); }
 	catch (e) { return null; }
-	if (stats.isFile()) {
-		const ext = PATH.extname(path).toLowerCase();
 
-		// Only add files with the provided extensions
-		if (options && 
-			options.extensions && 
-			options.extensions.length && 
-			options.extensions.indexOf(ext) === -1)
-			return null;
+	// Skip if it matches the exclude regex
+	if (options && options.exclude && options.exclude.test(path))
+		return null;  
 
+	if (stats.isFile()) {
+		// Skip if it does not match the include regex
+		if (options && options.includeOnly && !options.includeOnly.test(path))
```

Comment:
So I like the idea of having an include only, but at what level do you decide to include it. 
Say for example you have a structure

`root/something/else/more/myfile.file`

`root/something/else/more2/myfile2.file`

You want to `includeOnly` the folder `more2`. The problem arises when the plugin will get to `root/something/`, which will not match `more2`.  Which will cause it to ignore the entire subtree. The only way I think we can implement this is to check the regex at the last directory or file level. 
What do you think?